frame/support/src/dispatch.rs

```diff
@@ -2834,6 +2834,22 @@ macro_rules! decl_module {
 			}
 		}
 
+		// Implement GetCallIndex for the Call.
+		impl<$trait_instance: $trait_name $(<I>, $instance: $instantiable)?> $crate::dispatch::GetCallIndex
+			for $call_type<$trait_instance $(, $instance)?> where $( $other_where_bounds )*
+		{
+			//fake impl
+			fn get_call_index(&self) -> u8 {
+				*self as u8
+			}
```

Comment:
IMHO it would be fine to the trait impl for the old `decl_module!` syntax.  
another incentive to switch over :man_shrugging:   

Do you think this is acceptable @kianenigma, @KiChjang ?

Reply:
Mmm, wouldn't it result in a compiler error? I think it's best that we remove the deck syntax altogether.

Reply:
We're almost 2 years into macros v2, I would either remove the `decl_*`, or stop backporting to it.

Reply:
(assuming it is anything beyond a non-negligible amount of effort to backport)

Reply:
Yes looks like it, unless you find an easy way to implement it in the old syntax :cry:.  
Although you can do half of it now: Just introduce `GetCallIndex` for the current call syntax and ignore it for the old `decl` macros.  
Then it won't be possible to use in `GetCallMetadata` but AFAIK it is still enough to use it in the current syntax.

Reply:
This `decl_module` macro overwhelms me a little bit. I don't see an easy way to port `call_index` taking into consideration that indices should be also implicitly set (exactly the same way) otherwise this would introduce breaking change.

I suggest to take more pragmatic path and wait for `decl_*` macros deprecation.

I'll update commit according to what you suggested in comment.

Reply:
>I suggest to take more pragmatic path and wait for decl_* macros deprecation.

yes this will be done (hopefully soon).